# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/worklet/tests/file_basic.html
# Commit: 4bed3a0e2332
# Full Hash: 4bed3a0e2332f24ffe38f08e234c8898c42249ee
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2016-11-07 02:21:38
# Regressor Bug: 1290021
# File Overlap Count: 1
# Description:
#   Bug 1290021 - Implement a prototype version of Houdini "Worklets Level 1" spec - part 1 - WebIDL interface, r=smaug
# ==============================================================================

diff -r d84108f190ee -r 4bed3a0e2332 dom/worklet/tests/file_basic.html
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/worklet/tests/file_basic.html	Sun Nov 06 09:53:34 2016 +0100
@@ -0,0 +1,33 @@
+<!DOCTYPE HTML>
+<html>
+<head>
+  <title>Test for Worklet</title>
+  <script type="application/javascript" src="common.js"></script>
+</head>
+<body>
+
+<script type="application/javascript">
+
+setupTest();
+
+var worklet = window.createWorklet();
+ok(!!worklet, "We have a Worklet");
+
+worklet.import("common.js")
+.then(() => {
+  ok(true, "Import should load a resource.");
+})
+.then(() => {
+  return worklet.import("404.js");
+})
+.then(() => {
+  ok(false, "The loading should fail.");
+}, () => {
+  ok(true, "The loading should fail.");
+}).then(() => {
+  SimpleTest.finish();
+});
+
+</script>
+</body>
+</html>